omegaconf/base.py

```diff
@@ -598,6 +620,22 @@ def _maybe_resolve_interpolation(
         key: Any,
         value: "Node",
         throw_on_resolution_failure: bool,
+    ) -> Any:
+        return self._maybe_resolve_interpolation_impl(
+            parent=parent,
+            key=key,
+            value=value,
+            throw_on_resolution_failure=throw_on_resolution_failure,
+            memo=set(),
```

Comment:
Shouldn't `memo` contain `id(self)` here?

Reply:
I think it's not a big deal because the consequence is that you may wait an extra step before triggering the exception. I see no reason not to have it though, do you?
If we wanted to test it we'd need to check which key triggers the cycle detection.

Reply:
Going to resolve this one since I don't think it matters much to delay the detection by one step